packages/fluent-theme/src/components/sendbox/Attachments.tsx

```diff
@@ -0,0 +1,163 @@
+import React, {
+  type ChangeEventHandler,
+  type DragEventHandler,
+  type ReactNode,
+  useCallback,
+  useEffect,
+  useRef,
+  useState
+} from 'react';
+import { hooks } from 'botframework-webchat-api';
+import { ToolbarButton } from './Toolbar';
+import { AttachmentIcon } from '../../icons/AttachmentIcon';
+import { AddDocumentIcon } from '../../icons/AddDocumentIcon';
+import { useStyles } from '../../styles';
+
+const { useLocalizer } = hooks;
+
+const styles = {
+  'webchat__sendbox__add-attachment': {
+    display: 'grid'
+  },
+
+  'webchat__sendbox__add-attachment-input': {
+    fontSize: 0,
+    width: 0,
+    height: 0
+  },
+
+  webchat__sendbox__attachment: {
+    border: '1px solid var(--colorNeutralStroke1)',
+    padding: '6px 8px',
+    borderRadius: 'var(--borderRadiusLarge)',
+    cursor: 'default',
+    width: 'fit-content'
+  },
+
+  'webchat__sendbox__attachment-dropzone': {
+    position: 'absolute',
+    inset: '0',
+    backgroundColor: 'var(--colorNeutralBackground4)',
+    display: 'grid',
+    placeItems: 'center',
+    placeContent: 'center',
+    gap: '8px',
+    borderRadius: 'inherit',
+    cursor: 'copy'
+  },
+
+  'webchat__sendbox__attachment-dropzone-icon': {
+    width: '36px',
+    height: '36px'
+  }
+};
+
+export function AddAttachmentButton(props: {
+  readonly disabled?: boolean | undefined;
+  readonly icon?: ReactNode | undefined;
+  readonly onFilesAdded: ((files: File[]) => void) | undefined;
+}) {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const classnames = useStyles(styles);
+  const localize = useLocalizer();
+
+  const handleClick = useCallback(() => {
+    inputRef.current?.click();
+  }, []);
+
+  const handleFileChange = useCallback<ChangeEventHandler<HTMLInputElement>>(({ target: { files } }) => {
+    if (files) {
+      props.onFilesAdded?.(Array.from(files));
+    }
+
+    if (inputRef.current) {
+      inputRef.current.value = '';
+    }
+  }, []);
+
+  return (
+    <div className={classnames['webchat__sendbox__add-attachment']}>
+      <input
```

Comment:
Try to componentize the `<input type="file">` into its own component. I.e. without style or icon or whatever. Just the business logic around the element.

Take it as a challenge of building an independent OSS project.

Reply:
Will look into it later. Not sure this should be priority now. Let me know if you want changes to be made in the PR